#! /bin/sh /usr/share/dpatch/dpatch-run
## 90_125-dates.dpatch by  <seneca@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixing textual date display

@DPATCH@

diff -Naur ledgersmb-1.2.5-new/LedgerSMB/Locale.pm ledgersmb-1.2.5-dirty/LedgerSMB/Locale.pm
--- ledgersmb-1.2.5-new/LedgerSMB/Locale.pm	2007-05-17 11:26:32.000000000 -0400
+++ ledgersmb-1.2.5-dirty/LedgerSMB/Locale.pm	2007-05-28 18:02:25.000000000 -0400
@@ -1,3 +1,4 @@
+
 =head1 NAME
 
 LedgerSMB::Locale  Locale handling class for LedgerSMB
@@ -9,17 +10,24 @@
 =head1 METHODS
 
 =item get_handle ($language_code)
+
 Returns a locale handle for accessing the other methods.  Inherited from 
 Locale::Maketext.
 
 =item text ($string)
+
 Returns the translation for the given string.  This is a legacy wrapper that
 merely calls $self->maketext.
 
 =item date ($myconfig, $date, $longformat)
-Returns the given date after formatting it.  If $longformat is set, the date 
-will be givin in the form of "_('September') 23 2007".  If $longformat is not
-set, the date will be output in the format specified by $myconfig->{dateformat}.
+
+Returns the given date after formatting it.  $longformat is a ternary flag that
+determines how the date is formatted.  If $longformat is true, the date will be
+given in the form of "_('September') 23 2007".  If $longformat is false but
+defined, the date will be in the form of "_('Sep') 23 2007" unless the date is
+given in the form 'yyyy.mm.dd', in which case it is returned as-is.  If
+$longformat is not defined, the date will be output in the format specified by
+$myconfig->{dateformat}.
 
 =head1 Copyright (C) 2006, The LedgerSMB core team.
 
@@ -60,6 +68,7 @@
 =cut
 
 package LedgerSMB::Locale;
+
 use base 'Locale::Maketext';
 use Locale::Maketext::Lexicon;
 use Encode;
@@ -84,8 +93,8 @@
 sub date {
     my ( $self, $myconfig, $date, $longformat ) = @_;
 
-##    my @longmonth = (qw(Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec));
-    my @longmonth = (
+    my @longmonth = (qw(Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec));
+    @longmonth = (
         "January",   "February", "March",    "April",
         "May ",      "June",     "July",     "August",
         "September", "October",  "November", "December"
